xdsl/transforms/apply_pdl.py

```diff
@@ -0,0 +1,67 @@
+from dataclasses import dataclass
+from io import StringIO
+
+from xdsl.context import MLContext
+from xdsl.dialects import builtin, pdl
+from xdsl.interpreter import Interpreter
+from xdsl.interpreters.experimental.pdl import (
+    PDLMatcher,
+    PDLRewriteFunctions,
+)
+from xdsl.ir import Operation, Region, SSAValue
+from xdsl.parser import Parser
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import PatternRewriter
+
+
+@dataclass(frozen=True)
+class ApplyPDLPass(ModulePass):
+    name = "apply-pdl"
+
+    pdl_file: str | None = None
```

Comment:
Could you please add a test that specifies the file explicitly, as well?

Reply:
Ok I put the extra file in the filecheck directory as well ?

Reply:
Yes, please!